sdk/go/common/workspace/plugins.go

```diff
@@ -2278,7 +2378,7 @@ func getCandidateExtensions() []string {
 // pluginRegexp matches plugin directory names: pulumi-KIND-NAME-VERSION.
 var pluginRegexp = regexp.MustCompile(
 	"^(?P<Kind>[a-z]+)-" + // KIND
-		"(?P<Name>[a-zA-Z0-9-]*[a-zA-Z0-9])-" + // NAME
+		"(?P<Name>[a-zA-Z0-9-_.]*[a-zA-Z0-9])-" + // NAME
```

Comment:
We're allowing dots to start names now?

Reply:
I meant to only allow it within the name for URLs.  E.g. github.com/pulumi/provider-bla gets turned into `github.com_pulumi_provider-bla`.  Not sure we should even allow `-` at the beginning of the name, but since we always did I guess it's fine.

Reply:
Hrm expressing not allowing `.` at the start of the regex is pretty awkward now.  `[a-zA-Z0-9-][a-zA-Z0-9-_.]*[a-zA-Z0-9])-` would force at least two character names which we didn't.  Though maybe that's fine, I doubt any providers with just a single character name exist currently anyway.